src/iarc7_motion/iarc_tasks/hold_position_task.py

```diff
@@ -0,0 +1,160 @@
+            elif not (self._height_checker.above_min_maneuver_height(
+                        self._drone_odometry.pose.pose.position.z)):
+                return (TaskAborted(msg='Drone is too low'),)
+            elif not (self._check_max_error()):
+                return (TaskAborted(msg='Position holder is not for translating'),)
+
+            if self._canceled:
+                return (TaskCanceled(),)
+
+            if not self._check_max_start_hold_range():
+                return (TaskAborted(msg='The provided hold position is too far'),)
+
+            # p-controller
+            x_vel_target = ((self._x_position - self._drone_odometry.pose.pose.position.x)
+                                * self._K_X)
+            y_vel_target = ((self._y_position - self._drone_odometry.pose.pose.position.y) 
+                                * self._K_Y)
+            z_vel_target = ((self._z_position - self._drone_odometry.pose.pose.position.z) 
+                                * self._K_Z)
+
+            #caps velocity
+            vel_target = math.sqrt(x_vel_target**2 + y_vel_target**2)
+
+            if vel_target > self._MAX_TRANSLATION_SPEED:
+                x_vel_target = x_vel_target * (self._MAX_TRANSLATION_SPEED/vel_target)
+                y_vel_target = y_vel_target * (self._MAX_TRANSLATION_SPEED/vel_target)
+            
+            if (abs(z_vel_target) > self._MAX_Z_VELOCITY):
+                z_vel_target = z_vel_target/abs(z_vel_target) * self._MAX_Z_VELOCITY
+
+            velocity = TwistStamped()
+            velocity.header.frame_id = 'level_quad'
+            velocity.header.stamp = rospy.Time.now()
+            velocity.twist.linear.x = x_vel_target
+            velocity.twist.linear.y = y_vel_target
+            velocity.twist.linear.z = z_vel_target
+            
+            return (TaskRunning(), VelocityCommand(velocity)) 
+            
+    def cancel(self):
+        rospy.loginfo('HoldPosition Task canceled')
+        self._canceled = True
+
+    def _check_max_error(self):
+        x_vel_target = (self._x_position - self._drone_odometry.pose.pose.position.x)
+        y_vel_target = (self._y_position - self._drone_odometry.pose.pose.position.y)
+        z_vel_target = (self._z_position - self._drone_odometry.pose.pose.position.z)
+
+        _distance_to_point = math.sqrt(x_vel_target**2 + y_vel_target**2 + z_vel_target**2)
+        
+        return (_distance_to_point <= self._MAX_RANGE)
+
+    def _check_max_start_hold_range(self):
+        x_vel_target = (self._x_position - self._drone_odometry.pose.pose.position.x)
+        y_vel_target = (self._y_position - self._drone_odometry.pose.pose.position.y)
+        z_vel_target = (self._z_position - self._drone_odometry.pose.pose.position.z)
+
+        _distance_to_point = math.sqrt(x_vel_target**2 + y_vel_target**2 + z_vel_target**2)
+        
+        return (_distance_to_point <= self._MAX_RANGE)
+
+    def _check_inputs(self):
```

Comment:
You don't need to do both of the checks on z since min maneuver height should always be positive and parameter checking should be done elsewhere to ensure that the parameter is positive.

Also I don't think you need a function for this because you're only checking one variable.